### PR TITLE
Fix path parameter validation

### PR DESCRIPTION
**Pull request by raymondbutcher**
This change, along with the updates to fleek-parser and fleek-router (thanks for merging them so quickly!), gives us working path parameter validation.

### CODE REVIEW

File: lib/validate/ctx.js [hunk 1]
```diff
@@ -32,7 +33,7 @@ module.exports = function (ctx, parameters) {
     switch (parameter.in) {
       case 'query' : src = request.query || {}; break;
       case 'body'  : src = request.body || {}; break;
-      case 'path'  : src = request.params || {}; break;
+      case 'path'  : src = requestParams || {}; break;
```
Comment (johnhof) [inline lib/validate/ctx.js]: I may be missing something, but I'm not totally sure what this is for. shouldn't it already default?

File: lib/validate/ctx.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ var _               = require('lodash');
 //
 module.exports = function (ctx, parameters) {
   var request = (ctx ||{}).request;
```
Comment (johnhof) [inline lib/validate/ctx.js]: maybe this should be `var request = (ctx ||{}).request || {};`?

**Comment by johnhof** [comment]
@raymondbutcher thanks for the contributions! we fell into a bit of a fixed use case, so its hard for us to find errors/improvements

File: lib/validate/ctx.js (hunk 1 above)
Comment (raymondbutcher) [inline lib/validate/ctx.js]: Koa-router sets URL parameters in `ctx.params` and not `ctx.request.params`. See https://github.com/alexmingoia/koa-router#url-parameters

Perhaps I should rename `requestParams` to `urlParameters` or `pathParameters` or something. What do you think?

**Comment by raymondbutcher** [comment]
@johnhof you're welcome! I figured your use case didn't include URL parameter validation when I tried it out and found it didn't work. But the fleekjs project as a whole is really close to what we want to use for a new API project, so thanks!

File: lib/validate/ctx.js (hunk 2 above)
Comment (raymondbutcher) [inline lib/validate/ctx.js]: I didn't look to see when or if this ever gets called without valid context, but I guess it wouldn't hurt. I'll update it.

**Comment by raymondbutcher** [comment]
@johnhof I've updated the branch with these key changes:
* implemented your suggestion of `var request = (ctx ||{}).request || {};`
* removed my extra variable, instead accessing `ctx` directly in the `switch`
* added tests
* **TEMPORARILY** depend on `git://github.com/fleekjs/fleek-parser.git` because the latest version on NPM doesn't have the path fix, somehow

**Comment by johnhof** [comment]
@raymondbutcher  sorry about that, my local wasnt up to date when I published. I'll update package.json after the merge before I publish. Thanks again!

**Comment by raymondbutcher** [comment]
That's quite alright. Cheers!

**Pull request merged**